### PR TITLE
Show what other users are playing on the server

### LINKED ISSUE #769: Show What Other Users Are Playing
It is very nice to see what other users are playing on the server.  Perhaps as a list on the left pane?

### PR DESCRIPTION
**Pull request by Rishabh-malhotraa**
closes #769

shows what other users are playing in the activity menu.

> Hummm.... I took a look in the server code, and it seems that you bumped into a bug. This endpoint was created in 2016 and I never looked into it again after a lot of refactorings, and it seems to not be working properly.... To complete your code, try to mock the response with more info.
>
> In terms of design, maybe add a header, "Now playing", then you don't need to add "current playing" on every line. I think it is important to show Song and Artist and the time (minutesAgo). The Album name is not necessary. We should also truncate long names with ellipsis (...).

Based on feedback, requested changed

|dialog box| dialog Box II|
|-|-|
| Screenshot from 2021-04-13 14-13-39 | Screenshot from 2021-04-13 14-14-01 |

### CODE REVIEW

File: ui/src/layout/ActivityPanel.js [hunk 1]
```diff
@@ -81,7 +114,28 @@ const ActivityPanel = () => {
...
+                  album: user.album,
+                  artist: user.artist,
+                  minutesAgo: user.minutesAgo || 0,
+                }
+              }) || []
+            )
+          }
+        })
+    }
+  }, [dispatch, open])
```
Comment (Rishabh-malhotraa) [inline ui/src/layout/ActivityPanel.js:138]: the useEffect runs whenever open changes it state where open = Boolean(anchorEl) i.e whenever we either open/close the modal but we don't want to make an API call when we close the modal hence this check condtion.

File: ui/src/layout/ActivityPanel.js [hunk 2]
```diff
@@ -42,6 +49,31 @@ const useStyles = makeStyles((theme) => ({
...
     minWidth: '15em',
   },
+  heading: {
+    padding: '0px 16px',
+  },
+  list: {
+    maxWidth: '300px',
+    textOverflow: 'ellipsis',
+    ' & .primary': {
+      '& .MuiListItemText-primary': {
```
Comment (Rishabh-malhotraa) [inline ui/src/layout/ActivityPanel.js:59]: this means for add following CSS to `.MuiListItemText-primary` wrapped around div with `class = "primary"`

File: ui/src/layout/ActivityPanel.js [hunk 3]
```diff
@@ -150,6 +204,37 @@ const ActivityPanel = () => {
...
+                    className="primary"
+                    primary={`${user.username}`}
+                    secondary={`${user.title} - ${user.artist}`}
+                  />
+                  <ListItemSecondaryAction>
+                    <ListItemText
+                      primary={`${user.minutesAgo} ${translate(
+                        'activity.minutesAgo'
+                      )}`}
+                      secondary="place-holder"
```
Comment (Rishabh-malhotraa) [inline ui/src/layout/ActivityPanel.js:231]: we need to add a "placeholder" to not break material UI CSS, i.e if we don't add a placeholder than css breaks for material UI like the following. we add visibility hidden to the div which contains the "place-holder" text 

Screenshot from 2021-04-13 15-00-27

Comment (Rishabh-malhotraa) [inline ui/src/layout/ActivityPanel.js:231]: we need to add a "placeholder" to not break material UI CSS, i.e if we don't add a placeholder than css breaks for material UI like the following.
Screenshot from 2021-04-13 15-00-27

 we then hide the placeholder text by adding visibility hidden to it. 

``` js
    '& .MuiListItemSecondaryAction-root': {
      '& .MuiListItemText-secondary': {
        visibility: 'hidden',
      },
    },

```

**Comment by metalheim** [comment]
Looks great IMHO :)

**Comment by deluan** [comment]
Sorry for not give the proper attention to this PR, but I have to fix the `nowPlaying` endpoint before enabling this in the UI. Have to find the time... :/

**Comment by deluan** [comment]
The `nowPlaying` endpoint is now fixed in master, I'll try to review and merge this for next release

**Comment by Rishabh-malhotraa** [comment]
> The `nowPlaying` endpoint is now fixed in master, I'll try to review and merge this for next release

great looking forward to it 🙌

File: ui/src/layout/ActivityPanel.js [hunk 4]
```diff
@@ -81,7 +117,28 @@ const ActivityPanel = () => {
           dispatch(scanStatusUpdate(data.scanStatus))
         }
       })
-  }, [dispatch])
+    if (open) {
+      fetchUtils
+        .fetchJson(subsonic.url('getNowPlaying'))
```
Comment (deluan) [inline ui/src/layout/ActivityPanel.js:122]: Please create a new function in `subsonic/index.js` for this.

File: ui/src/layout/ActivityPanel.js [hunk 5]
```diff
@@ -150,6 +207,37 @@ const ActivityPanel = () => {
...
+                  <ListItemAvatar>
+                    <Avatar
+                      alt={`${user.title} ${translate('activity.coverArt')}`}
+                      src={subsonic.getCoverArtUrl(user)}
+                    />
+                  </ListItemAvatar>
+                  <ListItemText
+                    className="primary"
+                    primary={`${user.username}`}
+                    secondary={`${user.title} - ${user.artist}`}
```
Comment (deluan) [inline ui/src/layout/ActivityPanel.js:227]: Can we add a link to the album here?

**Review by deluan — CHANGES_REQUESTED** [review]

**Comment by deluan** [comment]
Hey @Rishabh-malhotraa, I just tried it in my local and looks great, with only a couple of issues (see the review comments).
Also, can we make the `title - artist` lines wrap around to up to two lines? The width of the popup is not enough for most songs:

Screen Shot 2021-06-25 at 1 29 36 PM

You'll need to pull the latest master...

**Comment by Rishabh-malhotraa** [comment]
> Hey @Rishabh-malhotraa, I just tried it in my local and looks great, with only a couple of issues (see the review comments).
> Also, can we make the `title - artist` lines wrap around to up to two lines? The width of the popup is not enough for most songs:
> 
> Screen Shot 2021-06-25 at 1 29 36 PM
> 
> You'll need to pull the latest master...

Thank you for the detailed review @deluan, I uninstalled ubuntu from my laptop on which I built navidrome, besides that I am quite busy preparing for coding interviews, regardless I'll try to build navidrome on window and get this done whenever I get some spare time.

**Comment by deluan** [comment]
No rush. By the way, it may be easier to use WSL insteado of trying to setup de dev environment directly on Windows..

**Comment by Rishabh-malhotraa** [comment]
> Also, can we make the `title - artist` lines wrap around to up to two lines? The width of the popup is not enough for most songs:

Rather than having wrap it around upto two lines what do you think about moving text from text to right, I checked on youtube music and they implement moving text on long albums names too

**Comment by deluan** [comment]
@Rishabh-malhotraa, yeah that looks good!

**Pull request closed without merging**